src/psyclone/tests/domain/lfric/lfric_integer_builtins_test.py

```diff
@@ -1110,3 +1110,23 @@ def test_real_X(tmpdir, monkeypatch, annexed, dist_mem):
         if not annexed:
             output_dm_2 = output_dm_2.replace("dof_annexed", "dof_owned")
         assert output_dm_2 in code
+
+
+def test_real_X_precision(tmpdir, monkeypatch):
+    '''Test that the builtin picks up and creates correct code for a
+    scalar with precision that is not the default i.e. not r_def.
+
+    '''
+    _, invoke_info = parse(os.path.join(BASE_PATH,
+                                        "15.28.2_real_X_builtin.f90"),
+                           api=API)
+    psy = PSyFactory(API).create(invoke_info)
+    # Test string method
+    first_invoke = psy.invokes.invoke_list[0]
+    kern = first_invoke.schedule.children[0].loop_body[0]
+    monkeypatch.setattr(kern.args[0], "_precision", "r_solver")
+    code = str(psy.gen)
+    assert ("USE constants_mod, ONLY: r_solver, i_def" in code)
```

Comment:
pylint isn't going to like these parentheses (and on the next line too).

Reply:
Oops.